pkg/utils/context.go

```diff
@@ -0,0 +1,13 @@
+package utils
+
+import (
+	"context"
+	"time"
+)
+
+// TODO: Timeout should be configurable.
+const deadline time.Duration = time.Second * 30
+
+func WithContext(ctx context.Context) (context.Context, context.CancelFunc) {
```

Comment:
Can you explain what this method is doing? It takes a `context.Context` but I don't see it being used.

Reply:
Never mind, I just got through the rest of the code and saw that it was copied here. I think that's a good idea, but for it to work we should pass in the `ctx` instead of creating a new context.

Reply:
And also maybe we should change the name of the method to better reflect what it's doing (adding a deadline to a context).

Reply:
Name changed